### DISCUSSION TITLE
[node] ArrayBuffer methods and properties

### DISCUSSION THREAD

**Discussion by ghost** [DefinitelyTyped/DefinitelyTyped#68989] [Issues with a @types package] [answered]
There are some methods and properties available for ArrayBuffer in NodeJS 20, for example `ArrayBuffer.resize()` and `ArrayBuffer.resizable`) plus  new options in the constructor. [See MDN table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).

I don't think the [**@types/node**](https://www.npmjs.com/package/@types/node) includes those, and I do have the latest in `package.json`:

```json
  "devDependencies": {
    "@types/node": "^20.11.27"
  }
```

Is this an error ? Are these queued to be added ? I don't see any issues related.

I wouldn't mind to add them but maybe this is done automatically and there is an error elsewhere.

[a comment by an automated account is not shown]

**Comment by yoursunny** [comment, marked as answer]
ArrayBuffer.prototype.resize is a feature of ES2024 language spec.
Thus, this does not belong to \@types/node.
Instead, it should be part of TypeScript itself, enabled via `"lib": "es2024"` or `"target": "es2024"`.

https://github.com/microsoft/TypeScript/issues/54636
